crypto/crypto.go

```diff
@@ -43,11 +46,19 @@ func GenerateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(S256(), rand.Reader)
 }
 
+// ToECDSAPub create ecdsa.PublicKey object by byte array. If pub length equals 64, then need inserts one byte in front
 func ToECDSAPub(pub []byte) *ecdsa.PublicKey {
 	if len(pub) == 0 {
 		return nil
 	}
-	x, y := elliptic.Unmarshal(S256(), pub)
+	if len(pub) == 65 {
+		x, y := elliptic.Unmarshal(S256(), pub)
+		return &ecdsa.PublicKey{Curve: S256(), X: x, Y: y}
+	}
+	pubNew := make([]byte, 65)
+	pubNew[0] = 4
```

Comment:
If 4 is a meaningful value, suggest to define a const variable with meaningful name.

Reply:
ok